diff --git a/libgambatte/Makefile.libretro b/libgambatte/Makefile.libretro
index 37c389d..c476324 100644
--- a/libgambatte/Makefile.libretro
+++ b/libgambatte/Makefile.libretro
@@ -132,8 +132,7 @@ ifeq ($(DEBUG), 1)
 	CFLAGS += -O0 -g
 	CXXFLAGS += -O0 -g
 else
-	CFLAGS += -O3
-	CXXFLAGS += -O3 -fno-exceptions -fno-rtti -DHAVE_STDINT_H
+	CXXFLAGS += -fno-exceptions -fno-rtti -DHAVE_STDINT_H
 endif
 
 RESAMPLER_DIR := ../common/resample/src
