diff --git a/Makefile b/Makefile
index f31ff77..6cab082 100644
--- a/Makefile
+++ b/Makefile
@@ -300,7 +300,7 @@ ifeq ($(HAVE_CG), 1)
    ifeq ($(OSX), 1)
       LIBS += -framework Cg
    else
-      LIBS += -lCg -lCgGL
+      LIBS += -I/opt/nvidia-cg-toolkit/include -lCg -lCgGL
    endif
 endif
 
@@ -483,8 +483,8 @@ install: $(TARGET)
 	rm -f $(OBJDIR)/git_version.o
 	mkdir -p $(DESTDIR)$(PREFIX)/bin 2>/dev/null || /bin/true
 	mkdir -p $(DESTDIR)$(GLOBAL_CONFIG_DIR) 2>/dev/null || /bin/true
-	mkdir -p $(DESTDIR)$(PREFIX)/share/man/man1 2>/dev/null || /bin/true
-	mkdir -p $(DESTDIR)$(PREFIX)/share/pixmaps 2>/dev/null || /bin/true
+	mkdir -p $(DESTDIR)/usr/share/man/man1 2>/dev/null || /bin/true
+	mkdir -p $(DESTDIR)/usr/share/pixmaps 2>/dev/null || /bin/true
 	install -m755 $(TARGET) $(DESTDIR)$(PREFIX)/bin 
 	install -m755 tools/cg2glsl.py $(DESTDIR)$(PREFIX)/bin/retroarch-cg2glsl
 	install -m644 retroarch.cfg $(DESTDIR)$(GLOBAL_CONFIG_DIR)/retroarch.cfg
@@ -492,8 +492,8 @@ install: $(TARGET)
 	install -m644 docs/retroarch-cg2glsl.1 $(DESTDIR)$(MAN_DIR)
 	install -m644 docs/retroarch-joyconfig.1 $(DESTDIR)$(MAN_DIR)
 	install -m644 docs/retrolaunch.1 $(DESTDIR)$(MAN_DIR)
-	install -m644 media/retroarch.png $(DESTDIR)$(PREFIX)/share/pixmaps
-	install -m644 media/retroarch.svg $(DESTDIR)$(PREFIX)/share/pixmaps
+	install -m644 media/retroarch.png $(DESTDIR)/usr/share/pixmaps
+	install -m644 media/retroarch.svg $(DESTDIR)/usr/share/pixmaps
 
 uninstall:
 	rm -f $(DESTDIR)$(PREFIX)/bin/retroarch
diff --git a/qb/config.libs.sh b/qb/config.libs.sh
index 1719b6d..8ea1e08 100644
--- a/qb/config.libs.sh
+++ b/qb/config.libs.sh
@@ -160,6 +160,12 @@ if [ "$HAVE_OPENGL" != 'no' ] && [ "$HAVE_GLES" != 'yes' ]; then
       check_lib CG "-framework Cg" cgCreateContext
    else
       # On some distros, -lCg doesn't link against -lstdc++ it seems ...
+      if [ -d /usr/lib64 ]; then
+          [ -d /opt/nvidia-cg-toolkit/lib64 ] && add_library_dirs /opt/nvidia-cg-toolkit/lib64
+      else
+          [ -d /opt/nvidia-cg-toolkit/lib ] && add_library_dirs /opt/nvidia-cg-toolkit/lib
+      fi
+      [ -d /opt/nvidia-cg-toolkit/include ] && add_include_dirs /opt/nvidia-cg-toolkit/include
       check_lib_cxx CG -lCg cgCreateContext
    fi
 else
diff --git a/qb/qb.params.sh b/qb/qb.params.sh
index b23d91d..464d4c0 100644
--- a/qb/qb.params.sh
+++ b/qb/qb.params.sh
@@ -39,7 +39,7 @@ EOF
 opt_exists() # $opt is returned if exists in OPTS
 {	opt=$(echo "$1" | tr '[a-z]' '[A-Z]')
 	for OPT in $OPTS; do [ "$opt" = "$OPT" ] && return; done
-	print_help; exit 1
+	print_help; echo "Unknown option $1"
 }
 
 parse_input() # Parse stuff :V
@@ -67,7 +67,7 @@ parse_input() # Parse stuff :V
 				eval "$opt=\"$val\""
 			;;
 			-h|--help) print_help; exit 0;;
-			*) print_help; exit 1;;
+			*) print_help; echo "Unknown option $1";;
 		esac
 		shift
 	done
