diff --git a/Makefile b/Makefile
index fd46ec1..0f80e18 100644
--- a/Makefile
+++ b/Makefile
@@ -150,8 +150,8 @@ install: $(TARGET)
 	rm -f $(OBJDIR)/git_version.o
 	mkdir -p $(DESTDIR)$(PREFIX)/bin 2>/dev/null || /bin/true
 	mkdir -p $(DESTDIR)$(GLOBAL_CONFIG_DIR) 2>/dev/null || /bin/true
-	mkdir -p $(DESTDIR)$(PREFIX)/share/man/man1 2>/dev/null || /bin/true
-	mkdir -p $(DESTDIR)$(PREFIX)/share/pixmaps 2>/dev/null || /bin/true
+	mkdir -p $(DESTDIR)/usr/share/man/man1 2>/dev/null || /bin/true
+	mkdir -p $(DESTDIR)/usr/share/pixmaps 2>/dev/null || /bin/true
 	install -m755 $(TARGET) $(DESTDIR)$(PREFIX)/bin 
 	install -m755 tools/cg2glsl.py $(DESTDIR)$(PREFIX)/bin/retroarch-cg2glsl
 	install -m644 retroarch.cfg $(DESTDIR)$(GLOBAL_CONFIG_DIR)/retroarch.cfg
@@ -159,8 +159,8 @@ install: $(TARGET)
 	install -m644 docs/retroarch-cg2glsl.1 $(DESTDIR)$(MAN_DIR)
 	install -m644 docs/retroarch-joyconfig.1 $(DESTDIR)$(MAN_DIR)
 	install -m644 docs/retrolaunch.1 $(DESTDIR)$(MAN_DIR)
-	install -m644 media/retroarch.png $(DESTDIR)$(PREFIX)/share/pixmaps
-	install -m644 media/retroarch.svg $(DESTDIR)$(PREFIX)/share/pixmaps
+	install -m644 media/retroarch.png $(DESTDIR)/usr/share/pixmaps
+	install -m644 media/retroarch.svg $(DESTDIR)/usr/share/pixmaps
 
 uninstall:
 	rm -f $(DESTDIR)$(PREFIX)/bin/retroarch
diff --git a/qb/config.libs.sh b/qb/config.libs.sh
index 6f58ad5..8a77f60 100644
--- a/qb/config.libs.sh
+++ b/qb/config.libs.sh
@@ -219,6 +219,15 @@ if [ "$HAVE_OPENGL" != 'no' ] && [ "$HAVE_GLES" != 'yes' ]; then
       [ "$HAVE_CG" = 'yes' ] && CG_LIBS='-lCg -lCgGL'
    fi
 
+   # Gentoo installs to /opt
+   if [ "$HAVE_CG" = 'no' ]; then
+       check_pkgconf CG nvidia-cg-toolkit
+       # Setup scripts don't work with how check_pkgconf would set it up, just fix them here
+       if [ "$HAVE_CG" = 'yes' ]; then
+           CG_LIBS='$($PKG_CONF_PATH --cflags) -lCg -lCgGL'
+       fi
+   fi
+
    # fix undefined variables
    PKG_CONF_USED="$PKG_CONF_USED CG"
 else
diff --git a/qb/qb.params.sh b/qb/qb.params.sh
index b23d91d..464d4c0 100644
--- a/qb/qb.params.sh
+++ b/qb/qb.params.sh
@@ -39,7 +39,7 @@ EOF
 opt_exists() # $opt is returned if exists in OPTS
 {	opt=$(echo "$1" | tr '[a-z]' '[A-Z]')
 	for OPT in $OPTS; do [ "$opt" = "$OPT" ] && return; done
-	print_help; exit 1
+	print_help; echo "Unknown option $1"
 }
 
 parse_input() # Parse stuff :V
@@ -67,7 +67,7 @@ parse_input() # Parse stuff :V
 				eval "$opt=\"$val\""
 			;;
 			-h|--help) print_help; exit 0;;
-			*) print_help; exit 1;;
+			*) print_help; echo "Unknown option $1";;
 		esac
 		shift
 	done
