diff --git a/Makefile b/Makefile
index 8134251..e55f6f4 100644
--- a/Makefile
+++ b/Makefile
@@ -143,16 +143,16 @@ install: $(TARGET)
 	rm -f $(OBJDIR)/git_version.o
 	mkdir -p $(DESTDIR)$(PREFIX)/bin 2>/dev/null || /bin/true
 	mkdir -p $(DESTDIR)$(GLOBAL_CONFIG_DIR) 2>/dev/null || /bin/true
-	mkdir -p $(DESTDIR)$(PREFIX)/share/man/man1 2>/dev/null || /bin/true
-	mkdir -p $(DESTDIR)$(PREFIX)/share/pixmaps 2>/dev/null || /bin/true
+	mkdir -p $(DESTDIR)/usr/share/man/man1 2>/dev/null || /bin/true
+	mkdir -p $(DESTDIR)/usr/share/pixmaps 2>/dev/null || /bin/true
 	install -m755 $(TARGET) $(DESTDIR)$(PREFIX)/bin 
 	install -m755 tools/cg2glsl.py $(DESTDIR)$(PREFIX)/bin/retroarch-cg2glsl
 	install -m644 retroarch.cfg $(DESTDIR)$(GLOBAL_CONFIG_DIR)/retroarch.cfg
 	install -m644 docs/retroarch.1 $(DESTDIR)$(MAN_DIR)
 	install -m644 docs/retroarch-cg2glsl.1 $(DESTDIR)$(MAN_DIR)
 	install -m644 docs/retroarch-joyconfig.1 $(DESTDIR)$(MAN_DIR)
-	install -m644 media/retroarch.png $(DESTDIR)$(PREFIX)/share/pixmaps
-	install -m644 media/retroarch.svg $(DESTDIR)$(PREFIX)/share/pixmaps
+	install -m644 media/retroarch.png $(DESTDIR)/usr/share/pixmaps
+	install -m644 media/retroarch.svg $(DESTDIR)/usr/share/pixmaps
 
 uninstall:
 	rm -f $(DESTDIR)$(PREFIX)/bin/retroarch
diff --git a/Makefile.common b/Makefile.common
index aa3d636..6f7ac50 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -489,6 +489,7 @@ ifeq ($(HAVE_CG), 1)
 	DEFINES += -DHAVE_CG
    OBJ += gfx/shader/shader_gl_cg.o
    LIBS += $(CG_LIBS)
+   CFLAGS += $(CG_CFLAGS)
 endif
 
 ifeq ($(HAVE_D3D9), 1)
diff --git a/qb/config.libs.sh b/qb/config.libs.sh
index fc82e7b..5b6d4a9 100644
--- a/qb/config.libs.sh
+++ b/qb/config.libs.sh
@@ -207,16 +207,21 @@ if [ "$OS" = 'Win32' ]; then
 fi
 
 if [ "$HAVE_OPENGL" != 'no' ] && [ "$HAVE_GLES" != 'yes' ]; then
-   if [ "$OS" = 'Darwin' ]; then
-      check_lib CG "-framework Cg" cgCreateContext
-      [ "$HAVE_CG" = 'yes' ] && CG_LIBS='-framework Cg'
-   elif [ "$OS" = 'Win32' ]; then
-      check_lib_cxx CG -lcg cgCreateContext
-      [ "$HAVE_CG" = 'yes' ] && CG_LIBS='-lcg -lcgGL'
-   else
-      # On some distros, -lCg doesn't link against -lstdc++ it seems ...
-      check_lib_cxx CG -lCg cgCreateContext
-      [ "$HAVE_CG" = 'yes' ] && CG_LIBS='-lCg -lCgGL'
+   # Gentoo installs to /opt, but provides pc
+   check_pkgconf CG nvidia-cg-toolkit-gl
+
+   if [ "$HAVE_CG" = 'no' ]; then
+      if [ "$OS" = 'Darwin' ]; then
+         check_lib CG "-framework Cg" cgCreateContext
+         [ "$HAVE_CG" = 'yes' ] && CG_LIBS='-framework Cg'
+      elif [ "$OS" = 'Win32' ]; then
+         check_lib_cxx CG -lcg cgCreateContext
+         [ "$HAVE_CG" = 'yes' ] && CG_LIBS='-lcg -lcgGL'
+      else
+         # On some distros, -lCg doesn't link against -lstdc++ it seems ...
+         check_lib_cxx CG -lCg cgCreateContext
+         [ "$HAVE_CG" = 'yes' ] && CG_LIBS='-lCg -lCgGL'
+      fi
    fi
 
    # fix undefined variables
diff --git a/qb/qb.params.sh b/qb/qb.params.sh
index b23d91d..464d4c0 100644
--- a/qb/qb.params.sh
+++ b/qb/qb.params.sh
@@ -39,7 +39,7 @@ EOF
 opt_exists() # $opt is returned if exists in OPTS
 {	opt=$(echo "$1" | tr '[a-z]' '[A-Z]')
 	for OPT in $OPTS; do [ "$opt" = "$OPT" ] && return; done
-	print_help; exit 1
+	print_help; echo "Unknown option $1"
 }
 
 parse_input() # Parse stuff :V
@@ -67,7 +67,7 @@ parse_input() # Parse stuff :V
 				eval "$opt=\"$val\""
 			;;
 			-h|--help) print_help; exit 0;;
-			*) print_help; exit 1;;
+			*) print_help; echo "Unknown option $1";;
 		esac
 		shift
 	done
