diff --git a/Makefile.libretro b/Makefile.libretro
index 410a45e..2e9f172 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -144,13 +144,13 @@ else
 endif
 
 ifeq ($(DEBUG), 1)
-   CFLAGS += -O0 -g
+   CFLAGS += -g
 else ifeq ($(platform),qnx)
-	CFLAGS += -Os -DNDEBUG
+	CFLAGS += -DNDEBUG
 else ifeq ($(platform), emscripten)
-	CFLAGS += -O2 -DNDEBUG
+	CFLAGS += -DNDEBUG
 else
-	CFLAGS += -O3 -DNDEBUG
+	CFLAGS += -DNDEBUG
 endif
 	
 LIBRETRO_SRC := $(GENPLUS_SRC_DIR)/genesis.c \
