diff --git a/libretro/Makefile b/libretro/Makefile
index 9705db8..2eb3513 100644
--- a/libretro/Makefile
+++ b/libretro/Makefile
@@ -131,14 +131,14 @@ endif
 all: $(TARGET)
 
 ifeq ($(DEBUG), 1)
-   CFLAGS += -O0 -g
-   CXXFLAGS += -O0 -g
+   CFLAGS += -g
+   CXXFLAGS += -g
 else ifeq ($(platform), ps3)
    CFLAGS += -O2
    CXXFLAGS += -O2
 else
-   CFLAGS += -O3
-   CXXFLAGS += -O3
+   CFLAGS +=
+   CXXFLAGS +=
 endif
 
 OBJS = $(NST_DIR)/source/core/NstZlib.o
